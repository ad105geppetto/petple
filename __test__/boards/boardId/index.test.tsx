import { props, mocks } from "./boardDetail.mocks";
import { fireEvent, render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import BoardDetailUI from "../../../src/components/units/board/detail/BoardDetail.presenter";
import BoardDetail from "../../../src/components/units/board/detail/BoardDetail.container";
import { useRouter } from "next/router";

describe("자유게시판 상세페이지 Unit Test & Integration Test", () => {
  it("게시판 상세페이지 스냅샷 테스트", () => {
    const mockedUseRouter = jest.spyOn(require("next/router"), "useRouter");
    mockedUseRouter.mockImplementation(() => ({
      query: { boardId: "6409e1391182750028ee7891" },
    }));

    const result = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <BoardDetailUI {...props} />
      </MockedProvider>
    );

    expect(result.container).toMatchSnapshot();
  });

  it("'목록으로' 버튼 클릭시, 페이지를 이동합니다.", () => {
    const router = {
      push: jest.fn(),
      query: { boardId: "6409e1391182750028ee7891" },
    };
    (useRouter as jest.Mock).mockReturnValue(router);

    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <BoardDetail />
      </MockedProvider>
    );

    const onClickList = screen.getByText("목록으로");
    fireEvent.click(onClickList);

    expect(router.push).toHaveBeenCalled();
  });

  it("'수정하기' 버튼 클릭시, 페이지를 이동합니다.", () => {
    const router = {
      push: jest.fn(),
      query: { boardId: "6409e1391182750028ee7891" },
    };
    (useRouter as jest.Mock).mockReturnValue(router);

    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <BoardDetail />
      </MockedProvider>
    );

    const onClickList = screen.getByText("수정하기");
    fireEvent.click(onClickList);

    expect(router.push).toHaveBeenCalled();
  });
});
