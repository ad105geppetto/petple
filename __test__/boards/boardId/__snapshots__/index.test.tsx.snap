// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`자유게시판 상세 페이지 Unit Test 게시판 상세페이지 스냅샷 테스트 1`] = `
<div>
  <div
    class="css-1cg5es3"
  >
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="css-13r7esh"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  class="css-13urh8h"
                />
                <div
                  class="css-1uhgteg"
                >
                  이미지 없음
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <div
    class="css-jhpxfm"
  >
    <img
      class="css-1n3uoyk"
      src="/profile.webp"
    />
    <div
      class="css-1nxvubt"
    >
      <div
        class="css-m2817q"
      >
        test
      </div>
      <div
        class="css-uahybi"
      >
        2023.02.24
      </div>
    </div>
  </div>
  <h1
    class="css-ljf5wq"
  >
    test
  </h1>
  <div
    class="css-eu34vv"
  >
    test
  </div>
  <div
    class="css-tjw2n1"
  >
    <div
      class="css-1sqyb6c"
    >
      목록으로
    </div>
    <div
      class="css-1sqyb6c"
    >
      수정하기
    </div>
  </div>
  
  <div
    class="css-1didpv3"
  >
    <div
      class="ant-space css-dev-only-do-not-override-1me4733 ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
      >
        <span
          aria-label="comment"
          class="anticon anticon-comment"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="comment"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M573 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40zm-280 0c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
            />
            <path
              d="M894 345a343.92 343.92 0 00-189-130v.1c-17.1-19-36.4-36.5-58-52.1-163.7-119-393.5-82.7-513 81-96.3 133-92.2 311.9 6 439l.8 132.6c0 3.2.5 6.4 1.5 9.4a31.95 31.95 0 0040.1 20.9L309 806c33.5 11.9 68.1 18.7 102.5 20.6l-.5.4c89.1 64.9 205.9 84.4 313 49l127.1 41.4c3.2 1 6.5 1.6 9.9 1.6 17.7 0 32-14.3 32-32V753c88.1-119.6 90.4-284.9 1-408zM323 735l-12-5-99 31-1-104-8-9c-84.6-103.2-90.2-251.9-11-361 96.4-132.2 281.2-161.4 413-66 132.2 96.1 161.5 280.6 66 412-80.1 109.9-223.5 150.5-348 102zm505-17l-8 10 1 104-98-33-12 5c-56 20.8-115.7 22.5-171 7l-.2-.1A367.31 367.31 0 00729 676c76.4-105.3 88.8-237.6 44.4-350.4l.6.4c23 16.5 44.1 37.1 62 62 72.6 99.6 68.5 235.2-8 330z"
            />
            <path
              d="M433 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
            />
          </svg>
        </span>
      </div>
    </div>
    <h5
      class="css-1d79vht"
    >
      댓글
    </h5>
  </div>
  <div
    class="css-dnt2qy"
  >
    <label
      class="css-9l9b0g"
      for="writer"
    >
      작성자
    </label>
    <input
      autocomplete="off"
      class="css-1hodvp8"
      id="writer"
      type="text"
      value=""
    />
    <label
      class="css-9l9b0g"
      for="password"
    >
      비밀번호
    </label>
    <input
      autocomplete="off"
      class="css-1hodvp8"
      id="password"
      type="password"
      value=""
    />
  </div>
  <div
    class="css-151eokt"
  >
    <input
      autocomplete="off"
      class="css-y52zy2"
      type="text"
      value=""
    />
    <div
      class="css-u7h1i6"
    >
      0
       / 100
    </div>
    <button
      class="css-2w0jzu"
    >
      댓글 
      작성
    </button>
  </div>
  
</div>
`;

exports[`자유게시판 상세페이지 Unit Test 게시판 상세페이지 스냅샷 테스트 1`] = `
<div>
  <div
    class="css-1cg5es3"
  >
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="css-13r7esh"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  class="css-13urh8h"
                />
                <div
                  class="css-1uhgteg"
                >
                  이미지 없음
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <div
    class="css-jhpxfm"
  >
    <img
      class="css-1n3uoyk"
      src="/profile.webp"
    />
    <div
      class="css-1nxvubt"
    >
      <div
        class="css-m2817q"
      >
        test
      </div>
      <div
        class="css-uahybi"
      >
        2023.02.24
      </div>
    </div>
  </div>
  <h1
    class="css-ljf5wq"
  >
    test
  </h1>
  <div
    class="css-eu34vv"
  >
    test
  </div>
  <div
    class="css-tjw2n1"
  >
    <div
      class="css-1sqyb6c"
    >
      목록으로
    </div>
    <div
      class="css-1sqyb6c"
    >
      수정하기
    </div>
  </div>
  
  <div
    class="css-1didpv3"
  >
    <div
      class="ant-space css-dev-only-do-not-override-1me4733 ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
      >
        <span
          aria-label="comment"
          class="anticon anticon-comment"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="comment"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M573 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40zm-280 0c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
            />
            <path
              d="M894 345a343.92 343.92 0 00-189-130v.1c-17.1-19-36.4-36.5-58-52.1-163.7-119-393.5-82.7-513 81-96.3 133-92.2 311.9 6 439l.8 132.6c0 3.2.5 6.4 1.5 9.4a31.95 31.95 0 0040.1 20.9L309 806c33.5 11.9 68.1 18.7 102.5 20.6l-.5.4c89.1 64.9 205.9 84.4 313 49l127.1 41.4c3.2 1 6.5 1.6 9.9 1.6 17.7 0 32-14.3 32-32V753c88.1-119.6 90.4-284.9 1-408zM323 735l-12-5-99 31-1-104-8-9c-84.6-103.2-90.2-251.9-11-361 96.4-132.2 281.2-161.4 413-66 132.2 96.1 161.5 280.6 66 412-80.1 109.9-223.5 150.5-348 102zm505-17l-8 10 1 104-98-33-12 5c-56 20.8-115.7 22.5-171 7l-.2-.1A367.31 367.31 0 00729 676c76.4-105.3 88.8-237.6 44.4-350.4l.6.4c23 16.5 44.1 37.1 62 62 72.6 99.6 68.5 235.2-8 330z"
            />
            <path
              d="M433 421c-23.1 0-41 17.9-41 40s17.9 40 41 40c21.1 0 39-17.9 39-40s-17.9-40-39-40z"
            />
          </svg>
        </span>
      </div>
    </div>
    <h5
      class="css-1d79vht"
    >
      댓글
    </h5>
  </div>
  <div
    class="css-dnt2qy"
  >
    <label
      class="css-9l9b0g"
      for="writer"
    >
      작성자
    </label>
    <input
      autocomplete="off"
      class="css-1hodvp8"
      id="writer"
      type="text"
      value=""
    />
    <label
      class="css-9l9b0g"
      for="password"
    >
      비밀번호
    </label>
    <input
      autocomplete="off"
      class="css-1hodvp8"
      id="password"
      type="password"
      value=""
    />
  </div>
  <div
    class="css-151eokt"
  >
    <input
      autocomplete="off"
      class="css-y52zy2"
      type="text"
      value=""
    />
    <div
      class="css-u7h1i6"
    >
      0
       / 100
    </div>
    <button
      class="css-2w0jzu"
    >
      댓글 
      작성
    </button>
  </div>
  
</div>
`;
